app/buckets/routes.js

```diff
@@ -7,6 +7,5 @@ module.exports = [
   {name: 'new', pattern: '/buckets/new', view: views.new_bucket},
   {name: 'create', pattern: '/buckets/new', view: views.create_bucket, method: 'POST'},
   {name: 'details', pattern: '/buckets/:id', view: views.bucket_details},
-  {name: 'edit', pattern: '/buckets/:id/edit', view: views.bucket_edit},
-
+  {name: 'grant_user_access', method: 'POST', pattern: '/buckets/:id/grant_user_access', view: views.grant_user_access},
```

Comment:
This URL is a strange hybrid of REST and RPC. RESTful URLs should not really contain verbs (eg: grant), because the verb is the HTTP method (eg: POST).

A more RESTful URL might be `/buckets/:bucket_id/access` where you might POST a payload like `{user: user_id, access_level: 'readwrite'}`.

Reply:
Whether the frontend should have RESTful URLs, I'm not really sure.